src/app/crypto.service.ts

```diff
@@ -172,4 +173,38 @@ export class CryptoService {
 
     return bs58check.encode(prefixAndKey);
   }
+
+  // NOTE: Our ETH address uses the bitcion/bitclout derivation path, not the ETH path.
+  // This is ugly but we only do it because they're just deposit addresses and we couldn't
+  // backfill this data for existing users because we store the derived private key.
+  // A user can still easily import their account to an ETH client to recover any ETH
+  // by coping the seedHex from local storage.
+  //
+  // We aren't using ethereumjs-util to minimize bundle size and protect from vulnerabilities
+  // until LavaMoat is ready
+  //
+  // Reference implementation: https://github.com/ethereumjs/ethereumjs-util/blob/master/src/account.ts#L249
+  keychainToEthAddress(keychain: HDNode, network: Network): string {
+    // Get the uncompressed key
+    const ec = new EC('secp256k1');
+    const publicKeyEC = ec.keyFromPublic(keychain.publicKey, 'array');
+
+    // ETH uses the last 40 characters of the 64 byte SHA3 Keccak 256
+    const uncompressedKey = Buffer.from(publicKeyEC.getPublic(false, 'array').slice(1));
+    const ethAddress = new Keccak(256).update(uncompressedKey).digest('hex').slice(24);
```

Comment:
In the reference implementation you've linked to in the comment for this function, it says:
```
  // Only take the lower 160bits of the hash
  return keccak(pubKey).slice(-20)
```
^This code only takes 20 bytes while yours takes 40.  Am I reading that correctly? Can you explain what the difference is?

Reply:
40 hex characters is 20 bytes :)

Reply:
Great, makes sense!